compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -701,6 +728,21 @@ checkGroundType ty = do
     _ <- checkType ty KStar
     checkGroundType' ty
 
+checkExceptionType' :: MonadGamma m => Type -> m ()
+checkExceptionType' = \case
+    TGeneralError -> pure ()
+    TArithmeticError -> pure ()
+    TContractError -> pure ()
+    TCon qtcon -> do
+      _ <- inWorld (lookupException qtcon)
+      pure ()
+    ty -> throwWithContext (EExpectedExceptionType ty)
+
+checkExceptionType :: MonadGamma m => Type -> m ()
+checkExceptionType ty = do
+    checkType ty KStar
+    checkExceptionType' ty
```

Comment:
We also need a check that the exception definition is well-formed.

Reply:
Yes, there's a TODO in checkModule to add this. :-)